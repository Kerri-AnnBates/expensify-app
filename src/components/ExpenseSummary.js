import React from "react";
import { connect } from "react-redux";
import getExpenseTotal from "../selectors/expense-total";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => (
	<div>
		<p>Viewing {expenseCount} {expenseCount === 1 ? `expense` : `expenses`} totaling {numeral(expensesTotal / 100).format("$0,0.00")}</p>
	</div>
);

const mapStateToProps = (state) => {
	const expenses = selectExpenses(state.expenses, state.filters);

	return {
		expenseCount: expenses.length,
		expensesTotal: getExpenseTotal(expenses)
	}
}

export default connect(mapStateToProps)(ExpenseSummary);