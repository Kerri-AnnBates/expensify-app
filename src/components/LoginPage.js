import React from "react";

const LoginPage = () => (
	<div>
		<form>
			<button>Login</button>
		</form>
	</div>
)

export default LoginPage;

