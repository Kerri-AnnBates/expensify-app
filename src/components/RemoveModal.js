import React from "react";
import Modal from 'react-modal';

const RemoveModal = (props) => (
	<Modal isOpen={props.isOpen} className="modal">
		<div className="modal__title">
			<div className="content-container">
				<h3>Remove Expense</h3>
			</div>
		</div>
		<div className="modal__body">
			<div className="content-container">
				<p>Are you sure you want to remove the <span>{props.expense.description}</span> expense?</p>
				<button
					className="button"
					onClick={props.onRemove}
				>Yes</button>
				<button className="button button--secondary" onClick={props.handleCloseModal} >No</button>
			</div>
		</div>
	</Modal>
);

export default RemoveModal;