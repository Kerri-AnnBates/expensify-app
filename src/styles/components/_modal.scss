.ReactModalPortal>div {
	opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
	align-items     : center;
	background-color: rgba($dark-blue, 0.75) !important;
	display         : flex;
	justify-content : center;
	transition      : opacity 400ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
	opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
	opacity: 0;
}

.modal {
	max-width       : 30rem;
	background-color: $off-white;
	color           : #fff;
}

.modal__body {
	margin-bottom: $s-size;
	color        : $dark-grey;

	button {
		margin-right: $s-size;
	}

	span {
		font-weight: 700;
	}
}

.modal__title {
	font-size       : $font-size-large;
	background-color: $dark-blue;

	h3 {
		margin: 0;
	}
}