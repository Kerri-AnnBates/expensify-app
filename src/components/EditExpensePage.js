import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import Modal from 'react-modal';
import RemoveModal from "./RemoveModal";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

Modal.setAppElement("#app");

export class EditExpensePage extends React.Component {
	state = {
		isOpen: false
	}

	onSubmit = (expense) => {
		const id = this.props.expense.id;

		this.props.startEditExpense(id, expense);
		this.props.history.push("/");
	}

	onRemove = () => {
		const id = this.props.expense.id;

		this.props.startRemoveExpense(id);
		this.props.history.push("/");
	}

	handleOpenModal = () => {
		this.setState(() => {
			return {
				isOpen: true
			}
		});
	}

	handleCloseModal = () => {
		this.setState(() => {
			return {
				isOpen: false
			}
		});
	}

	render() {
		return (
			<div>
				<div className="page-header">
					<div className="content-container">
						<h1 className="page-header__title">Edit Expense</h1>
					</div>
				</div>
				<div className="content-container">
					<ExpenseForm
						expense={this.props.expense}
						onSubmit={this.onSubmit}
					/>
					<button onClick={this.handleOpenModal} className="button button--secondary">Remove Expense</button>
				</div>
				<RemoveModal
					isOpen={this.state.isOpen}
					expense={this.props.expense}
					onRemove={this.onRemove}
					handleCloseModal={this.handleCloseModal}
					onRequestClose={this.handleCloseModal}
					contentLabel="Remove expense"
				/>
			</div>
		)
	}
}

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find((expense) => expense.id === props.match.params.id)
	}
}

const mapDispatchToProps = (dispatch) => ({
	startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
	startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);