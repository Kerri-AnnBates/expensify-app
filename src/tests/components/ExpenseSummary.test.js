import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import expenses from "../fixtures/expenses";
import getExpenseTotal from "../../selectors/expense-total";

test("should render single expense summary", () => {
	const expense = [expenses[0]];
	const wrapper = shallow(
		<ExpenseSummary
			expenseCount={expense.length}
			expensesTotal={getExpenseTotal(expense)}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});

test("should render multiple expenses summary", () => {
	const wrapper = shallow(
		<ExpenseSummary
			expenseCount={expenses.length}
			expensesTotal={getExpenseTotal(expenses)}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});