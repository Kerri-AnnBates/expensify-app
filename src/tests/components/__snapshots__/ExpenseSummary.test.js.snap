// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple expenses summary 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totaling 
    $1,141.95
  </p>
</div>
`;

exports[`should render single expense summary 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totaling 
    $1.95
  </p>
</div>
`;
